src/cascade/executor/model_results_main.py

```diff
@@ -32,27 +73,35 @@ def _get_model_results(model_version_id, model_type):
         pandas.DataFrame: containing all columns in the "save results" upload table
     """
 
-    if model_type.upper() == "AT":
-        database = "dismod-at-dev"
-    elif model_type.upper() == "ODE":
-        database = "epi"
+    if db and table:
+        if db in dbs.keys() and table in tables.keys():
+            model_results_data = _retrieve_from_database(model_version_id, dbs[db], tables[table])
+            return model_results_data
+        else:
+            sys.exit(f"""Unknown db and table combination.  Acceptable databases are
+                     {dbs.keys()} and acceptable tables are {tables.keys()}""")
     else:
-        raise ValueError(f"model must be of type 'AT' or 'ODE', not {model_type}")
-
-    table = "epi.model_estimate_fit"
-
-    query = f"""
-    SELECT * FROM {table}
-    WHERE
-        model_version_id = %(model_version_id)s
-    """
-
-    with connection(database=database) as c:
-        model_results_data = pd.read_sql(query, c, params={"model_version_id": model_version_id})
-        CODELOG.debug(f"""Downloaded {len(model_results_data)} lines of dismod {model_type}
-                      data for model_version_id {model_version_id} from '{database}'""")
-
-    return model_results_data
+        # check all, if find more than one report it, else return one
+        model_results = None
+        model_results_empty = None
```

Comment:
This is new to me. In order to tell the reader that the dataframe is empty, the code assigns it to a variable with the name "empty". Think of this algorithm in terms of the minimum number of variables to define its state. There are two questions: what's available, and, if only one is available, what are its values? The function commits to retrieving everything it finds, so go with that. In short: let model_results be a list. Then count at the end.

Reply:
Thanks for commenting on this, because I didn't really like my final answer, but.
Here's what I was thinking.  If I keep non-empty's in model_results, then I can return the one non-empty one.  I can't, however, return an empty one (which would be something that has all the column names, but no data rows, which gets returned from the db retrieval).  If I keep empty's and non-empty's in model results, and there is only one non-empty in the list, I need to know which one it is.

Reply:
If an error is immediately raised to indicate multiple results, we can't report to them where all the results are, to help them choose one.